src/generate.js

```diff
@@ -258,81 +270,71 @@ export const generateCSSRuleset = (
-        ];
-        for (let j = 0; j < prefixedKeys.length; ++j) {
-            if (!originalElements.hasOwnProperty(prefixedKeys[j])) {
-                sortOrder[prefixedKeys[j]] = i - 0.5;
-                originalElements[prefixedKeys[j]] = originalElements[key];
-            }
-        }
-    }
+    const elementNames = Object.keys(prefixedElements);
+    if (elementNames.length !== handledDeclarations.keyOrder.length) {
+        // There are some prefixed values, so we need to figure out how to sort
+        // them.
+        //
+        // Loop through prefixedElements, looking for anything that is not in
+        // sortOrder, which means it was added by prefixAll. This means that we
+        // need to figure out where it should appear in the sortOrder.
+        for (let i = 0; i < elementNames.length; i++) {
+            if (!originalElements.hasOwnProperty(elementNames[i])) {
+                // This element is not in the sortOrder, which means it is a prefixed
+                // value that was added by prefixAll. Let's try to figure out where it
+                // goes.
+                let originalStyle;
+                if (elementNames[i][0] === 'W') {
+                    // This is a Webkit-prefixed style, like "WebkitTransition". Let's
+                    // find its original style's sort order.
+                    originalStyle = elementNames[i][6].toLowerCase() + elementNames[i].slice(7);
+                } else if (elementNames[i][1] === 'o') {
+                    // This is a Moz-prefixed style, like "MozTransition". We check
+                    // the second character to avoid colliding with Ms-prefixed
+                    // styles. Let's find its original style's sort order.
+                    originalStyle = elementNames[i][3].toLowerCase() + elementNames[i].slice(4);
+                } else { // if (elementNames[i][1] === 's') {
+                    // This is a Ms-prefixed style, like "MsTransition".
+                    originalStyle = elementNames[i][2].toLowerCase() + elementNames[i].slice(3);
+                }
 
-    // Calculate the sort order of a given property.
-    function sortOrderForProperty([key, value]) {
-        if (sortOrder.hasOwnProperty(key)) {
-            if (originalElements.hasOwnProperty(key) &&
-                    originalElements[key] !== value) {
-                // The value is prefixed. Sort this just before the key with
-                // the unprefixed value.
-                return sortOrder[key] - 0.25;
-            } else {
-                // Either the key and value are unprefixed here, or this is a
-                // prefixed key. Either way, this is handled by the sortOrder
-                // calculation above.
-                return sortOrder[key];
+                if (originalStyle && originalElements.hasOwnProperty(originalStyle)) {
+                    const originalIndex = handledDeclarations.keyOrder.indexOf(originalStyle);
```

Comment:
Yep, my benchmark included some things that are prefixed and some things that are not prefixed. This is faster for both because it gets all of the work done in fewer loops and avoids creating new objects in favor of mutating.